src/mqt/problemsolver/partialcompiler/Partial_QAOA.py

```diff
@@ -0,0 +1,192 @@
+        """
+        Creates a QAOA problem instance with a random number of known/offline edges and a random number of unknown/online edges.
+
+        :param num_qubits: Number of qubits in the problem instance
+        :param repetitions: Number of repetitions of the problem and mixer unitaries
+        """
+        self.num_qubits = num_qubits
+        self.repetitions = repetitions
+        self.offline_time_edges = []
+
+        # add offline edges
+        for i in range(1, num_qubits):
+            if np.random.random() < P_SAMPLE_TWO_QUBIT_GATE:
+                self.offline_time_edges.append((0, i))
+        self.problem_parameters: list[Parameter] = []
+        self.mapping: list[int] = []
+
+        # add online edges
+        self.online_time_edges = []
+        for i in range(1, num_qubits):
+            for j in range(i + 1, num_qubits):
+                if np.random.random() < P_SAMPLE_TWO_QUBIT_GATE:
+                    self.online_time_edges.append((i, j))
+
+        # determine device
+        manila_config = FakeManila().configuration()
+        montreal_config = FakeMontreal().configuration()
+        washington_config = FakeWashington().configuration()
+        if num_qubits <= manila_config.n_qubits:
+            self.coupling_map = manila_config.coupling_map
+            self.device_qubits = manila_config.n_qubits
+        elif num_qubits <= montreal_config.n_qubits:
+            self.coupling_map = montreal_config.coupling_map
+            self.device_qubits = montreal_config.n_qubits
+        elif num_qubits <= washington_config.n_qubits:
+            self.coupling_map = washington_config.coupling_map
+            self.device_qubits = washington_config.n_qubits
+
+    def get_uncompiled_circuit(
+        self, include_online_edges: bool = False
+    ) -> tuple[QuantumCircuit, list[QuantumCircuit], list[QuantumCircuit]]:
+        """Return the state preparation circuit and lists of problem and mixer circuits without any compilation.
+         Online edges are optionally included.
+
+        :param include_online_edges: Include online edges in the problem circuits
+        """
+        qc_prep = QuantumCircuit(self.num_qubits)
+        qc_prep.h(range(self.num_qubits))
+        qc_prep.barrier()
+        qcs_problem = []
+        qcs_mix = []
+        self.problem_parameters = []
+        for i in range(self.repetitions):
+            qc_problem = QuantumCircuit(self.num_qubits)
+            p = Parameter(f"a_{i}")
+            self.problem_parameters.append(p)
+            for elem in self.offline_time_edges:
+                qc_problem.rzz(p, elem[0], elem[1])
+            if include_online_edges:
+                for elem in self.online_time_edges:
+                    qc_problem.rzz(p, elem[0], elem[1])
+
+            qcs_problem.append(qc_problem)
+
+            m = Parameter(f"b_{i}")
+            qc_mix = QuantumCircuit(self.num_qubits)
+            qc_mix.barrier()
```

Comment:
shouldn't that barrier be part of `qc_problem`?

Reply:
In that case, the offline and online edges would be separated by a barrier and I thought it would be more intuitive to not separate those edges.

Reply:
Ah, that makes sense 👍🏼